### PR TITLE
#168050010 User Role Settings

### PR DESCRIPTION
**Pull request by didasmbalanya**
#### What does this PR do?
 - add functionality that allows the super admin to assign other roles
 - add tests to automate the tests for feature
 - change jwt payload for sign up and sign in feature to use email for consistency

#### Description of Task to be completed?
 - create roles model
 - create seeds for role models and seeds to create user super admin
 - set up functionality to enable super admin to change roles

#### How should this be manually tested?
 - use the `git fetch --all` to get this branch
 - 'git checkout ft-user-role-setting-168050010` to checkout to this branch
 - add an environment `SUPER_ADMIN_PASS` and a value, that will allow you to log in the super admin to get the token. with email `technitesdev@gmail.com` which you can get from the database after seed.
 - `npm run migrate:refresh` to undo and redo migrations
 -`npm run seed` to create the seeds on the tables. ie Superadmin
 - run server and user the `localhost:3000/api/v1/admin/users` on postman
 - add seeded super admins token in the header
 - after whuch you can use a put method with `{ new_role: "role value", email: "email of user to change role" }
` 
#### What are the relevant pivotal tracker stories?
[#168050010](https://www.pivotaltracker.com/story/show/168050010
Screen Shot 2019-09-17 at 23 33 52
)
#### Questions:

### CODE REVIEW

File: src/controllers/AdminController.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+   */
+  static async changeRole(req, res) {
+    const id = Number(req.params.id);
+    if (isNaN(id)) return res.status(400).send({ status: 400, error: 'Bad request', message: `ID param value '${req.params.id}' should be an integer` });
+    try {
+      const foundUser = await findUserById(id);
+      if (!foundUser) return res.status(404).send({ status: 404, error: `User with id ${req.params.id} does not exist` });
+      const oldRole = foundUser.role_value;
+      if (oldRole === req.body.new_role) return res.status(409).send({ status: 409, message: 'User already has the role', errpr: 'conflict' });
+      await database.User.update(
```
Comment (Jordybastien) [inline src/controllers/AdminController.js:23]: Personally I think it's better having this in Services for consistency

File: src/database/migrations/20190917071255-create-role.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-unused-vars */
+
+module.exports = {
```
Comment (Jordybastien) [inline src/database/migrations/20190917071255-create-role.js]: I think you should change this also to ES6 for consistency

File: src/database/models/role.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable no-unused-vars */
+
+module.exports = (sequelize, DataTypes) => {
```
Comment (Jordybastien) [inline src/database/models/role.js]: Here too

File: src/database/seeders/20190917101843-role-values.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-unused-vars */
+
+module.exports = {
```
Comment (Jordybastien) [inline src/database/seeders/20190917101843-role-values.js]: Same here

File: src/utils/AuthHelper.js [hunk 5]
```diff
@@ -16,10 +16,10 @@ class AuthHelper {
    */
   static jwtSign(payload, time = '24h') {
     // remove password from User Model
-    if ('password' in payload) {
-      const { password, ...patchedUser } = payload;
-      return jwt.sign(patchedUser, JWT_SECRET, { expiresIn: time });
-    }
+    // if (payload.password) {
```
Comment (Jordybastien) [inline src/utils/AuthHelper.js]: Commented out codes

**Comment by Jordybastien** [comment]
> Screen Shot 2019-09-17 at 23 33 52
> )
> 
> #### What are the relevant Github Issues?
> #### Screenshots (if appropriate)
> #### Questions:

The feature is well implemented, however i have an issue regarding the response i am receiving which is somehow different to the one in the PR description on the `newRole` field, as you can notice mine is null
didas

File: src/controllers/AdminController.js (hunk 1 above)
Comment (didasmbalanya) [reply src/controllers/AdminController.js:23]: how do you suggest I go about it?

Comment (Jordybastien) [reply src/controllers/AdminController.js:23]: I suggest you add it to the `UserServices` and create a method for doing those functionalities there.
PS: that's just my opinion as longs as it works its okay for me

**Review by gik-hub — DISMISSED** [review]
Awsome,.. everything will be set if you fix the slight errors I highlighted which are:
- trying to make a user a super admin which makes their role null
- I also expected a user to be changed role with their email as highlighted in the acceptance criteria

**Review by gik-hub — APPROVED** [review]
cool

**Review by Jordybastien — APPROVED** [review]

**Review by mucyomiller — APPROVED** [review]
LGTM

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**